rules/import-style.js

```diff
@@ -127,6 +128,175 @@ const defaultStyles = {
+				localName: specifier.local.name,
+				importedName: specifier.type === 'ImportDefaultSpecifier' ? 'default' : specifier.imported.name,
+			}));
+	} else if (node.type === 'VariableDeclarator' && node.id.type === 'ObjectPattern') {
+		importedNames = node.id.properties
+			.map(property => {
+				if (property.type === 'RestElement') {
+					return {
+						localName: property.argument.name,
+						importedName: undefined,
+					};
+				}
+
+				return {
+					localName: property.value.name,
+					importedName: property.key.name,
+				};
+			});
+	}
+
+	if (importedNames.length === 0) {
+		return;
+	}
+
+	const scope = sourceCode.getScope(node);
+
+	const namespaceIdentifier = getNamespaceIdentifier(moduleName);
+
+	// Check if any of the named imports match our desired namespace identifier
+	const hasMatchingNamedImport = importedNames.some(
+		({localName}) => localName === namespaceIdentifier,
+	);
+
+	// Only avoid capture if there's no matching named import
+	const uniqueNamespaceIdentifier = hasMatchingNamedImport
+		? namespaceIdentifier
+		: avoidCapture(namespaceIdentifier, [scope]);
+
+	// For VariableDeclarator, we need to handle the parent VariableDeclaration
+	const hasSemicolon = sourceCode.getText(
+		node.type === 'VariableDeclarator' ? node.parent : node,
+	).endsWith(';');
+
+	const importFix = fixer.replaceTextRange(
+		node.type === 'VariableDeclarator'
+			? [node.parent.range[0], node.parent.range[1]]
+			: [node.range[0], node.range[1]],
+		node.type === 'ImportDeclaration'
+			? `import * as ${uniqueNamespaceIdentifier} from ${sourceCode.getText(node.source)}${hasSemicolon ? ';' : ''}`
+			: `const ${uniqueNamespaceIdentifier} = require(${sourceCode.getText(
+				node.type === 'VariableDeclarator' ? node.init.arguments[0] : node.arguments[0],
+			)})${hasSemicolon ? ';' : ''}`,
+	);
+
+	if (hasMatchingNamedImport) {
+		return [importFix];
+	}
+
+	const referenceFixes = importedNames.flatMap(({localName, importedName}) => {
+		// Skip rest patterns since they should be kept as is
+		if (importedName === undefined) {
+			return [];
+		}
+
+		const programScope = sourceCode.getScope(sourceCode.ast);
+
+		const getAllReferences = scope => {
+			let references = scope.references.filter(
+				reference => reference.identifier.name === localName
+				// Skip the original declaration
+					&& !(node.type === 'VariableDeclarator' && reference.identifier === node.id)
+					&& !(node.type === 'VariableDeclarator' && node.id.type === 'ObjectPattern'
+						&& node.id.properties.some(p => p.value === reference.identifier)),
+			);
+
+			for (const childScope of scope.childScopes) {
+				references = [...references, ...getAllReferences(childScope)];
+			}
+
+			return references;
+		};
+
+		const references = getAllReferences(programScope);
+
+		return references.map(reference => fixer.replaceText(reference.identifier, `${uniqueNamespaceIdentifier}.${importedName}`),
+		);
+	});
+
+	return [importFix, ...referenceFixes];
```

Comment:
Please use a generator function instead return all at the end.

Reply:
Done 👍